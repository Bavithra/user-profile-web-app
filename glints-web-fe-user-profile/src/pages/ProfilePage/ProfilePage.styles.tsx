import styled from "styled-components";
import { FontSizes } from "../../styles/FontSizes";

import { mediaMax } from "../../styles/MediaQuery.styles";

export const Title = styled.h1`
  padding: 16px;
  margin: 16px;

  ${mediaMax("md")} {
    padding: 8px;
    margin: 8px;
    text-align: center;
    font-size: ${FontSizes.size4};
  }
`;

export const ProfilePageContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1 1 0%;

  ${mediaMax("md")} {
    flex-direction: column;
  }
`;

export const DetailsContainer = styled.div`
  width: 100%;
`;

export const InputContainer = styled.div`
  width: 100%;
  display: flex;
  grid-template-columns: 1fr 1fr;

  ${mediaMax("md")} {
    display: flex;
    flex-direction: column;
    padding: 5px;
  }
`;

export const HeaderContainer = styled.div`
  display: flex;

  ${mediaMax("md")} {
    display: initial;
  }
`;

export const SearchContainer = styled.div`
  height: 56px;
  width: 462px;

  ${mediaMax("md")} {
    height: 36px;
    width: 200px;
    flex-grow: 1;
  }
`;

export const Header = styled.div`
  display: flex;
  margin-bottom: 24px;
  justify-content: space-between;
  align-items: flex-start;
`;

export const FilterContainer = styled.div`
  display: flex;
  margin-bottom: 32px;
`;

export const MeetingButtonContainer = styled.div`
  margin-top: 12px;
  ${mediaMax("md")} {
    > div {
      width: 168px;
    }
  }
`;

export const ButtonContainer = styled.div`
  display: flex;
  ${mediaMax("md")} {
    margin: 0 0 10px 0;
    flex-direction: column-reverse;
  }
`;

export const PaginationContainer = styled.div`
  position: relative;

  ${mediaMax("lg")} {
    display: flex;
    flex-direction: column-reverse;
  }
`;

export const ItemsPerPageContainer = styled.div`
  position: absolute;
  top: 0;
  right: 0;

  ${mediaMax("lg")} {
    position: relative;
    align-self: flex-end;
  }
`;
