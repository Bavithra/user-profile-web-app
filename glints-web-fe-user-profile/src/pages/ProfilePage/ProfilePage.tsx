import React, { useState } from "react";

import ImageUpload from "../../common/ImageUpload";
import TextInput from "../../common/TextInput";

import WorkExperienceList from "../../components/WorkExperienceList";
import WorkExperienceAddModal from "../../components/WorkExperienceAddModal";

import { WorkExperience } from "../../types/WorkExperience.type";

import {
  DetailsContainer,
  InputContainer,
  ProfilePageContainer,
  Title,
} from "./ProfilePage.styles";
import WorkExperienceUtil from "../../utils/WorkExperienceUtil";
import Button from "../../common/Button";
import { Text } from "../../styles/Common.styles";

export default function ProfilePage() {
  const [name, setName] = useState<string>("");
  const [age, setAge] = useState<string>();
  const [selectedIndex, setSelectedIndex] = useState<number>(-1);

  const [isWorkExperienceModalOpen, setIsWorkExperienceModalOpen] =
    useState(false);

  const [workExperiences, setWorkExperiences] = useState<WorkExperience[]>([]);
  const [workExperience, setWorkExperience] = useState<WorkExperience>(
    WorkExperienceUtil.getInitialWorkExperienceInput()
  );

  function handleAddWorkExperienceClick(
    workExperience: WorkExperience,
    index: number
  ) {
    setWorkExperience(workExperience);
    setSelectedIndex(index);
    setIsWorkExperienceModalOpen(true);
  }

  function handleDeleteWorkExperienceClick(index: number) {
    const tempArray = workExperiences;
    tempArray.splice(index, 1);
    setSelectedIndex(index);
    setWorkExperiences(tempArray);
  }

  function handleLastNameChange(value: string) {
    setName(value);
  }

  function handleAgeChange(value: string) {
    setAge(value);
  }

  function handleProfileSaveClick() {
    //TODO: implement call to post api
  }

  function handleSaveButtonClick(workExperience: WorkExperience) {
    if (selectedIndex === -1) {
      const value = {
        ...workExperience,
        id: `${workExperience.company}-${workExperience["start-date"]}-${workExperience["end-date"]}}`,
      };
      setWorkExperiences([...workExperiences, value]);
    } else {
      const tempArray = workExperiences;
      tempArray[selectedIndex] = workExperience;
      setWorkExperiences(tempArray);
    }
  }

  return (
    <>
      {name.length > 0 && <Title> Hello {name}</Title>}
      <ProfilePageContainer>
        <div>
          <ImageUpload />
        </div>
        <DetailsContainer>
          <InputContainer>
            <TextInput
              label="Name*"
              value={name}
              onChange={handleLastNameChange}
            />
            <TextInput
              label="Age*"
              type="number"
              value={age}
              onChange={handleAgeChange}
            />
          </InputContainer>
          <Text>
            Please save all the changes you have made to the profile by clicking
            the below button
          </Text>

          <Button
            isDisable={name === "" || age === undefined || age === ''}
            onClick={handleProfileSaveClick}
          >
            SAVE PROFILE
          </Button>
          <WorkExperienceList
            workExperiences={workExperiences}
            onAddWorkExperienceClick={handleAddWorkExperienceClick}
            onDeleteWorkExperienceClick={handleDeleteWorkExperienceClick}
          />
        </DetailsContainer>

        <WorkExperienceAddModal
          isOpen={isWorkExperienceModalOpen}
          workExperience={workExperience}
          setIsWorkExperienceModalOpen={setIsWorkExperienceModalOpen}
          setWorkExperience={setWorkExperience}
          onSaveButtonClick={handleSaveButtonClick}
        />
      </ProfilePageContainer>
    </>
  );
}
