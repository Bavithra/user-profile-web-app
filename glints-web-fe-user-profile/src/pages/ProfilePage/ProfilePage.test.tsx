import React from "react";

import ProfilePage from "./ProfilePage";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

const mockStore = configureStore();
let store;

const initialState = {user:{
  user: {
    name: '',
    email: '',
    age: '',
    'profile-image': '',
    'work-experience': [],
  }
}}

describe("ProfilePage Block", () => {
  it("renders correctly", () => {
    store = mockStore(initialState);

    const tree = renderer
      .create(
        <Provider store={store}>
          <ProfilePage />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Check test render', () => {
  store = mockStore(initialState);

  it('Shows "Hello world!"', () => {
    store = mockStore(initialState)
    const { getByText } = render(<Provider store={store}><ProfilePage /></Provider>)

    expect(getByText('Please remember to save all the changes(name, image, email, age and work experience) you have made to the profile by clicking the below button')).not.toBeNull();
    expect(getByText('Add Work Experience')).not.toBeNull();
    expect(getByText('WORK EXPERIENCE')).not.toBeNull();
    expect(getByText('SAVE CHANGES')).not.toBeNull();
    expect(getByText('77.9% of employers surveyed consider work experience to be a crucial data point in job applications. So be sure to fill up this section to raise your chances of securing an interview!')).not.toBeNull();


  })
})
