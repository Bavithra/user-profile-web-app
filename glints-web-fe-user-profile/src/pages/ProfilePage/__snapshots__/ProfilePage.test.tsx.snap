// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage Block renders correctly 1`] = `
<div
  className="sc-bZkfAO lTGee"
>
  <div>
    <div
      className="sc-bczRLJ hZeqla"
    >
      <img
        alt=""
        className="sc-gsnTZi eioidg"
        src="avatar.png"
      />
      <button
        className="sc-hKMtZM lKwCO"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-xs "
          color="white"
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         Upload Image
      </button>
      <input
        accept="image/*"
        className="sc-dkzDqf tPwAK"
        data-testid="file-input"
        onChange={[Function]}
        type="file"
      />
    </div>
  </div>
  <div
    className="sc-kLLXSd bGXPbi"
  >
    <div
      className="sc-ikZpkk eKcSBs"
    >
      <div
        className="sc-gKXOVf iejHYC"
      >
        <input
          className="sc-ftvSup uFJpj"
          maxLength={200}
          min="0"
          onChange={[Function]}
          readOnly={false}
          type="string"
          value=""
        />
        <label
          className="sc-iBkjds dHKUjl"
        >
          Name*
        </label>
      </div>
      <div
        className="sc-gKXOVf iejHYC"
      >
        <input
          className="sc-ftvSup uFJpj"
          maxLength={200}
          min="0"
          onBlur={[Function]}
          onChange={[Function]}
          readOnly={false}
          type="string"
          value=""
        />
        <label
          className="sc-iBkjds dHKUjl"
        >
          Email*
        </label>
      </div>
      <div
        className="sc-gKXOVf iejHYC"
      >
        <input
          className="sc-ftvSup uFJpj"
          maxLength={200}
          min="0"
          onChange={[Function]}
          readOnly={false}
          type="number"
        />
        <label
          className="sc-iBkjds dHKUjl"
        >
          Age*
        </label>
      </div>
    </div>
    <div
      className="sc-jSMfEi hpfud"
    >
      Please save all the changes you have made to the profile by clicking the below button
    </div>
    <button
      className="sc-hKMtZM kJKxQG"
      disabled={true}
      onClick={[Function]}
    >
      SAVE CHANGES
    </button>
    <h2
      className="sc-fnykZs fgxqkd"
    >
      WORK EXPERIENCE
    </h2>
    <div
      className="sc-bjUoiL ehsFp"
    >
      77.9% of employers surveyed consider work experience to be a crucial data point in job applications. So be sure to fill up this section to raise your chances of securing an interview!
      <button
        className="sc-fEOsli dokDSl"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-circle-plus fa-sm "
          data-icon="circle-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 368C269.3 368 280 357.3 280 344V280H344C357.3 280 368 269.3 368 256C368 242.7 357.3 232 344 232H280V168C280 154.7 269.3 144 256 144C242.7 144 232 154.7 232 168V232H168C154.7 232 144 242.7 144 256C144 269.3 154.7 280 168 280H232V344C232 357.3 242.7 368 256 368z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Add Work Experience
      </button>
    </div>
  </div>
</div>
`;
